staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -201,6 +202,12 @@ func newRequestForProxy(location *url.URL, req *http.Request) (*http.Request, co
 	newReq.URL = location
 	newReq.Host = location.Host
 
+	// If the original request has an audit ID, let's make sure we propagate this
+	// to the aggregated server.
+	if auditID, found := genericapirequest.AuditIDFrom(req.Context()); found {
```

Comment:
have you checked `newReq.Header = utilnet.CloneHeader(req.Header)` ?

`req` should already have the `AuditID` header set by the filter

Reply:
If the user specify an audit ID only then `req` will already have the `Audit-ID` header. If we generate an audit ID then we save it in the context. 
if the context already has an audit ID then it has higher precedence, and we want to set it to the new `req`.

Reply:
okay, makes sense, thx.